cmd/soroban-rpc/internal/methods/get_events.go

```diff
@@ -293,34 +298,61 @@ func (s *SegmentFilter) UnmarshalJSON(p []byte) error {
 }
 
 type PaginationOptions struct {
-	Cursor *events.Cursor `json:"cursor,omitempty"`
-	Limit  uint           `json:"limit,omitempty"`
+	Cursor *db.Cursor `json:"cursor,omitempty"`
+	Limit  uint       `json:"limit,omitempty"`
 }
 
 type GetEventsResponse struct {
 	Events       []EventInfo `json:"events"`
 	LatestLedger uint32      `json:"latestLedger"`
 }
 
-type eventScanner interface {
-	Scan(eventRange events.Range, f events.ScanFunction) (uint32, error)
-}
-
 type eventsRPCHandler struct {
-	scanner      eventScanner
+	dbReader     db.EventReader
 	maxLimit     uint
 	defaultLimit uint
+	logger       *log.Entry
+	ledgerReader db.LedgerReader
 }
 
-func (h eventsRPCHandler) getEvents(request GetEventsRequest) (GetEventsResponse, error) {
+func combineContractIDs(filters []EventFilter) ([][]byte, error) {
+	contractIDSet := make(map[string]struct{})
+
+	for _, filter := range filters {
+		for _, contractID := range filter.ContractIDs {
+			contractIDSet[contractID] = struct{}{}
+		}
+	}
+
+	contractIDs := make([][]byte, 0, len(contractIDSet))
+	for contractID := range contractIDSet {
+		id, err := strkey.Decode(strkey.VersionByteContract, contractID)
+		if err != nil {
+			return nil, fmt.Errorf("contract ID %v invalid", contractID)
```

Comment:
```suggestion
			return nil, errors.Wrapf(err, "invalid contract ID: %v", contractID)
```

Reply:
Bad advice... modern Go says you should use `errors.Join(err, fmt.Errorf("contract ID invalid: %v", contractID))` here :roll_eyes:

Reply:
Yeah I think we just need to stick to one pattern either `errors.Wrap` (from our internal package) vs the one you mentioned. Also our linter does not like `errors.Wrap` :(